datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedPartitionAssigner.java

```diff
@@ -291,13 +274,17 @@ private void registerLoadBasedPartitionAssignmentMetrics(String datastream) {
   }
 
   void cleanupMetrics() {
-    _partitionAssignmentStatsMap.keySet().forEach(this::unregisterMetricsForDatastream);
-    _partitionAssignmentStatsMap.clear();
+    _metricsForDatastream.keySet().forEach(this::unregisterMetricsForDatastream);
   }
 
   void unregisterMetricsForDatastream(String datastream) {
```

Comment:
Can you double check if there is a test to cover this code path of registration and deregistration? My guess is no, otherwise, the test would have caught issue with the older patch. Can you add a test to cover that scenario?

Reply:
Can do.